staging/src/k8s.io/apiserver/pkg/util/proxy/proxy.go

```diff
@@ -96,8 +96,12 @@ func ResolveCluster(services listersv1.ServiceLister, namespace, id string, port
 	}
 
 	switch {
+	// use service domain for Headless Service
 	case svc.Spec.Type == v1.ServiceTypeClusterIP && svc.Spec.ClusterIP == v1.ClusterIPNone:
-		return nil, fmt.Errorf(`cannot route to service with ClusterIP "None"`)
+		return &url.URL{
+			Scheme: "https",
```

Comment:
The scheme here could be "http" as well. Better not hard-coded here.

Reply:
It is required to be https